src/main/java/ch/njol/skript/expressions/ExprPlayerStatistics.java

```diff
@@ -0,0 +1,194 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.EntityUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.Statistic;
+import org.bukkit.Statistic.Type;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.log.runtime.*;
+
+@Name("Player Statistics")
+@Description({
+	"Get or set the statistics of a player.",
+	"Some statistics require an entity type or item type to be specified. "
+		+ "For example, the 'kill entity' statistic requires an entity type. You can see more about this in "
+		+ "<a href='https://www.digminecraft.com/getting_started/statistics.php'>the DigMinecraft page about statistics</a>."
+})
+@Examples({
+	"set the mc kill entity statistic for a pig of player to 10",
+	"add 5 to the minecraft kill entity statistic for a pig of player",
+	"broadcast \"You have left the game %mc leave game statistic of player% times!\""
+})
+@Since("INSERT VERSION")
+public class ExprPlayerStatistics extends SimpleExpression<Integer> implements SyntaxRuntimeErrorProducer {
+
+	static {
+		Skript.registerExpression(ExprPlayerStatistics.class, Integer.class, ExpressionType.COMBINED,
+			"[the] (minecraft|mc) %statistic% [for %-entitydata/itemtype%] (from|of) %offlineplayers%",
+			"%offlineplayers%'[s] (minecraft|mc) %statistic% [for %-entitydata/itemtype%]"
+		);
+	}
+
+	private Expression<Statistic> statistic;
+	private Expression<OfflinePlayer> player;
+	private Expression<?> ofType;
+
+	private Node node;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 0) {
+			player = (Expression<OfflinePlayer>) exprs[2];
+			statistic = (Expression<Statistic>) exprs[0];
+			ofType = exprs[1];
+		} else {
+			player = (Expression<OfflinePlayer>) exprs[0];
+			statistic = (Expression<Statistic>) exprs[1];
+			ofType = exprs[2];
+		}
```

Comment:
An OR operator can be used to optimize this.
```suggestion
		player = (Expression<OfflinePlayer>) exprs[matchedPattern ^ 2];
		statistic = (Expression<Statistic>) exprs[matchedPattern];
		ofType = exprs[matchedPattern ^ 1];
```

Reply:
1 ^ 2 would return 3, not 0.

Reply:
i believe it's fine here, `1 ^ 1` is `0` and for ofType it's also fine